ttkwidgets/tickscale.py

```diff
@@ -0,0 +1,633 @@
+                            pady2 = pady
+                    else:
+                        pady1, pady2 = pady, pady
+
+        # ticks
+        pady1_2, pady2_2 = 0, 0
+        if self._tickinterval:
+            nb_interv = int(self._extent / self._tickinterval)
+            h = self.scale.winfo_reqheight()
+            if self._tickpos == 's':
+                for i in range(nb_interv + 1):
+                    tick = self._start + i * self._tickinterval
+                    self.ticks.append(tick)
+                    self.ticklabels.append(ttk.Label(self,
+                                                     style=self._style_name + ".TLabel",
+                                                     text=self._formatter.format(tick)))
+                    self.ticklabels[i].place(in_=self.scale, bordermode='outside',
+                                             x=0, y=h + pady2 + 1, anchor='n')
+                pady2_2 = self.ticklabels[-1].winfo_reqheight()
+            else:  # self._tickpos == 'n':
+                for i in range(nb_interv + 1):
+                    tick = self._start + i * self._tickinterval
+                    self.ticks.append(tick)
+                    self.ticklabels.append(ttk.Label(self,
+                                                     style=self._style_name + ".TLabel",
+                                                     text=self._formatter.format(tick)))
+                    self.ticklabels[i].place(in_=self.scale, bordermode='outside',
+                                             x=0, y=-1 - pady1, anchor='s')
+                pady1_2 = self.ticklabels[-1].winfo_reqheight()
+            self.update_idletasks()
+        self.scale.grid_configure(pady=(pady1 + pady1_2, pady2 + pady2_2),
+                                  padx=(padx1, padx2))
+
+    def _display_value_horizontal(self, value):
+        if self._showvalue:
+            self.label.configure(text=self._formatter.format(float(value)))
+            self.update_idletasks()
+            if self._labelpos in ['n', 's']:
+                # position (in pixel) of the center of the slider
+                x = self.convert_to_pixels(float(value))
+                # pay attention to the borders
+                half_width = self.label.winfo_width() / 2
+                if x + half_width > self.scale.winfo_width():
+                    x = self.scale.winfo_width() - half_width
+                elif x - half_width < 0:
+                    x = half_width
+                self.label.place_configure(x=x)
+
+    def _display_value_vertical(self, value):
+        """
+        Display the current value and update the label position.
+        Return the pady necessary to display the label.
+        """
+        if self._showvalue:
+            self.label.configure(text=self._formatter.format(float(value)))
+            if self._labelpos in ['e', 'w']:
+                y = self.convert_to_pixels(float(value))
+                self.label.place_configure(y=y)
```

Comment:
The docstring reports a return value, but there is no value actually returned. Is the docstring wrong?

Reply:
The function used to return something and I have forgotten to update the docstring after modifying the function.